### PR TITLE
Remove swift-support-tools-core

### PR DESCRIPTION
**Pull request by giginet**
related #3 

To remove SwiftPM dependencies, I'm going to drop swift-support-tools-core dependencies.

- Replace `TSCBasic.FileSystem` to original `FileSystem`
- Replace `TSCBasic.AbsolutePath` to `Foundation.URL`

Some implementations still use `TSCBasic`. they will be removed.

### CODE REVIEW

File: Sources/ScipioKit/ProjectGenerator.swift [hunk 1]
```diff
@@ -49,52 +50,50 @@ struct ProjectGenerator {
         embedDebugSymbols isDebugSymbolsEmbedded: Bool,
         frameworkType: FrameworkType
     ) throws -> Result {
-        let projectPath = package.projectPath
+        let projectPath = AbsolutePath(package.projectPath.path)
 
         let project = try pbxproj(
```
Comment (giginet) [inline Sources/ScipioKit/ProjectGenerator.swift:55]: `pbxproj` still needs `AbsolutePath` and `TSCBasic.FileSystem`. However, this calling will be removed.

File: Sources/ScipioKit/Cache/CacheSystem.swift [hunk 2]
```diff
@@ -214,6 +221,6 @@ struct CacheSystem {
 extension CacheKey {
     func calculateChecksum() throws -> String {
         let data = try jsonEncoder.encode(self)
-        return ByteString(data).sha256Checksum
+        return SHA256().hash(ByteString(data)).hexadecimalRepresentation
```
Comment (giginet) [inline Sources/ScipioKit/Cache/CacheSystem.swift:224]: same implementation

File: Sources/ScipioKit/FileSystem.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    func write(_ data: Data, to path: URL) -> Bool
+    func exists(_ path: URL) -> Bool
+    func contents(of path: URL) -> Data?
+    @discardableResult
+    func changeCurrentWorkingDirectory(to destination: URL) -> Bool
+    var currentWorkingDirectory: URL? { get }
+    func createDirectory(_ path: URL, recursive: Bool) throws
+    func copy(from source: URL, to destination: URL) throws
+    var cachesDirectory: URL? { get }
+    func removeFileTree(at path: URL) throws
```
Comment (mtj0928) [inline Sources/ScipioKit/FileSystem.swift:14]: (IMO)
The following order might be better for readability.
```swift
public protocol FileSystem {
    var currentWorkingDirectory: URL? { get }
    var cachesDirectory: URL? { get }

    @discardableResult
    func write(_ data: Data, to path: URL) -> Bool
    func exists(_ path: URL) -> Bool
    func contents(of path: URL) -> Data?
    @discardableResult
    func changeCurrentWorkingDirectory(to destination: URL) -> Bool
    func createDirectory(_ path: URL, recursive: Bool) throws
    func copy(from source: URL, to destination: URL) throws
    func removeFileTree(at path: URL) throws
}
```

File: Sources/ScipioKit/FileSystem.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+public let localFileSystem: FileSystem = LocalFileSystem()
```
Comment (mtj0928) [inline Sources/ScipioKit/FileSystem.swift:17]:
```suggestion
public let localFileSystem: some FileSystem = LocalFileSystem()
```

**Review by mtj0928 — APPROVED** [review]
LGTM, with trivial comments

**Review by giuk-jung — APPROVED** [review]
LGTM

File: Sources/ScipioKit/Cache/CacheSystem.swift [hunk 5]
```diff
@@ -94,35 +95,38 @@ public struct CacheKey: Hashable, Codable, Equatable {
...
+        case couldNotReadVersionFile(URL)
 
         var errorDescription: String? {
             switch self {
             case .revisionNotDetected(let packageName):
                 return "Repository version is not detected for \(packageName)."
             case .compilerVersionNotDetected:
                 return "Compiler version not detected. Please check your environment"
+            case .couldNotReadVersionFile(let path):
+                return "Could not read VersionFile \(path.path)"
```
Comment (freddi-kit) [inline Sources/ScipioKit/Cache/CacheSystem.swift:121]: Error is thrown here https://github.com/giginet/Scipio/pull/9/files#diff-c95016fdd23c7e500c6ca14a1de27966f249347c35a9e51c9e9974bde9d69d94R161 but will be converted to `return false`. So I'm curious how this error message works well.

**Review by freddi-kit — APPROVED** [review]

File: Sources/ScipioKit/Cache/CacheSystem.swift (hunk 5 above)
Comment (giginet) [reply Sources/ScipioKit/Cache/CacheSystem.swift:121]: This error might not raise. So I do not care about it.

Comment (freddi-kit) [reply Sources/ScipioKit/Cache/CacheSystem.swift:121]: i see

**Comment by giginet** [comment]
Thanks!

**Pull request merged**